### DISCUSSION TITLE
I want to make a custom edge type of smoothstep having button on hover!

### DISCUSSION THREAD

**Discussion by shubh-am8** [xyflow/xyflow#2044] [Q&A] [answered]
hi, i had working on this flow from last 2 months when i make a custom edge to add some extra features to it (like button on hover, animation on hover, validation to connect only 1 node at a time, updateable edge)i was always stuck at a point to make it type of "smoothstep" because it always set at default type.
And without custom edge i am not able to get this happen.
So please help me out, i am eagerly waiting for your suggestions and ideas.

**Comment by dmahely-alef** [comment]
Hi @shubh-am8, so have you implemented a custom edge? And is your requirement to have the edge label be a button?

**Reply by shubh-am8** [reply]
yes

**Comment by smadapathi** [comment, marked as answer]
If I understand your requirements, you need to have custom edge with button on hover features etc. and you want to make that edge as smoothstep. If yes, then you can get `edgepath` from [utils](https://reactflow.dev/docs/api/edges/edge-utils/). Example code on [sandbox](https://codesandbox.io/s/sandpack-project-forked-pefrzy?file=/CustomEdge.js:2512-2523)

**Reply by shubh-am8** [reply]
Thank you so much
